### ISSUE TITLE
generateObject with AWS Bedrock (Claude 3.5 Sonnet) Fails with ValidationException

### ISSUE THREAD

**Issue by mbokinala** [vercel/ai#2166]
### Description

Calling `generateObject()` with the AWS Bedrock Claude 3.5 Sonnet model (`bedrock("anthropic.claude-3-5-sonnet-20240620-v1:0")`) causes the following error to occur:

```
/Users/mbokinala/projects/vly-v2/node_modules/@aws-sdk/client-bedrock-runtime/dist-cjs/index.js:1082
  const exception = new ValidationException({
                    ^

ValidationException: The format of the value at toolConfig.tools.0.toolSpec.inputSchema.json is invalid. Provide a json object for the field and try again.
    at de_ValidationExceptionRes (/Users/mbokinala/projects/vly-v2/node_modules/@aws-sdk/client-bedrock-runtime/dist-cjs/index.js:1082:21)
    at de_CommandError (/Users/mbokinala/projects/vly-v2/node_modules/@aws-sdk/client-bedrock-runtime/dist-cjs/index.js:937:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/mbokinala/projects/vly-v2/node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20
    at async /Users/mbokinala/projects/vly-v2/node_modules/@smithy/core/dist-cjs/index.js:165:18
    at async /Users/mbokinala/projects/vly-v2/node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38
    at async /Users/mbokinala/projects/vly-v2/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:34:22
    at BedrockChatLanguageModel.doGenerate (/Users/mbokinala/projects/vly-v2/node_modules/@ai-sdk/amazon-bedrock/src/bedrock-chat-language-model.ts:159:22)
    at _retryWithExponentialBackoff (/Users/mbokinala/projects/vly-v2/node_modules/ai/core/util/retry-with-exponential-backoff.ts:36:12)
    at generateObject (/Users/mbokinala/projects/vly-v2/node_modules/ai/core/generate-object/generate-object.ts:178:30) {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    requestId: '51c84dca-51b0-4487-83b2-956b83fd61a5',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  }
}

Node.js v18.19.1
```

### Code example
(taken from the [`generateObject` example](https://sdk.vercel.ai/docs/reference/ai-sdk-core/generate-object))
```ts
(async () => {
  const { object } = await generateObject({
    model: bedrock("anthropic.claude-3-5-sonnet-20240620-v1:0"),
    schema: z.object({
      recipe: z.object({
        name: z.string(),
        ingredients: z.array(z.string()),
        steps: z.array(z.string()),
      }),
    }),
    prompt: "Generate a lasagna recipe.",
  });

  console.log(object);
})();
```

### Additional context

This error does NOT occur when using the regular anthropic provider for the same model (`anthropic("claude-3-5-sonnet-20240620")`).

The error also does not occur when using Bedrock with `generateText()`

**Comment by alexpate** [comment]
I think there's something going on [here](https://github.com/vercel/ai/blob/main/packages/amazon-bedrock/src/bedrock-chat-language-model.ts#L132).

Passing `mode.tool.parameters` without stringifying it makes the request work, but breaks types.

```
inputSchema: { json: mode.tool.parameters },
```

**Comment by fahreziadh** [comment]
I get the same issue when using stream Object or generate Object, using bedrock with model claude 3.5 sonnet

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2252

**Comment by lgrammel** [comment]
Fix in https://github.com/vercel/ai/releases/tag/%40ai-sdk/amazon-bedrock%400.0.6

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete provider implementation failure in existing object-generation behavior, subsequently confirmed and fixed by the maintainers.

### Evidence
- `generateObject()` and `streamObject()` with Bedrock Claude 3.5 Sonnet return a 400 `ValidationException` because `inputSchema.json` has an invalid format.
- The same schema works with the Anthropic provider, and Bedrock `generateText()` is unaffected.
- Comments identify that the schema was incorrectly stringified, and a maintainer links PR #2252 and confirms the fix in `@ai-sdk/amazon-bedrock@0.0.6`.
